src/psfmachine/ffi.py

```diff
@@ -0,0 +1,924 @@
+"""Subclass of `Machine` that Specifically work with FFIs"""
+import os
+import numpy as np
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+
+from astropy.io import fits
+from astropy.stats import SigmaClip
+from astropy.time import Time
+from astropy.wcs import WCS
+from photutils import Background2D, MedianBackground, BkgZoomInterpolator
+
+# from . import PACKAGEDIR
+from .utils import do_tiled_query
+from .machine import Machine
+from .version import __version__
+
+__all__ = ["FFIMachine"]
+
+
+class FFIMachine(Machine):
+    """
+    Subclass of Machine for working with FFI data. It is a subclass of Machine
+    """
+
+    def __init__(
+        self,
+        channel=1,
+        quarter=5,
+        wcs=None,
+        limit_radius=32.0,
+        n_r_knots=10,
+        n_phi_knots=15,
+        cut_r=6,
+        rmin=1,
+        rmax=16,
+        **kwargs,
+    ):
+        """
+        Class to work with FFI data.
+
+        Parameters
+        ----------
+        channel : int
+            Channel number to be used
+        quarter : int
+            Quarter/Campagn nunmber to be used (for Kepler data).
+        wcs : astropy.wcs
+            World coordinates system solution for the FFI. Used for plotting.
+        **kwargs
+            Keyword attributes that contain information parsed from `from_file()` and
+            is used to initialize a `Machine` class object.
+
+        Attributes
+        ----------
+        All attributes inherited from Machine.
+
+        meta : dictionary
+            Meta data information related to the FFI
+        wcs : astropy.wcs
+            World coordinates system solution for the FFI. Used for plotting.
+        flux_2d : numpy.ndarray
+            2D image representation of the FFI, used for plotting.
+        channel : int
+            Channel number to be used
+        quarter : int
+            Quarter/Campagn nunmber to be used (for Kepler data).
+        """
+        self.column = kwargs["column"].ravel()
+        self.row = kwargs["row"].ravel()
+        self.ra = kwargs["ra"].ravel()
+        self.dec = kwargs["dec"].ravel()
+        # keep 2d image for easy plotting
+        self.flux_2d = kwargs["flux"]
+        # reshape flux and flux_err as [ntimes, npix]
+        self.flux = kwargs["flux"].reshape(kwargs["flux"].shape[0], -1)
+        self.flux_err = kwargs["flux_err"].reshape(kwargs["flux_err"].shape[0], -1)
+        self.sources = kwargs["sources"]
+
```

Comment:
Let's make sure to save the shape of the input flux somewhere (e.g. (1024x1100))